app/models/activity/import/field.rb

```diff
@@ -0,0 +1,78 @@
+class Activity::Import::Field
+  class << self
+    def all
+      from_source(source)
+    end
+
+    def find_by(attribute_name:)
+      field = source[attribute_name]
+      return nil if field.nil?
+
+      new(
+        attribute_name: attribute_name,
+        heading: field[:heading],
+        exclude_from_converter: field[:exclude_from_converter]
+      )
+    end
+
+    def find_all_by(headings:)
+      fields = source.select { |_key, value| headings.include?(value[:heading]) }
+
+      from_source(fields)
+    end
+
+    def where(level:, type:)
+      fields = source.select { |_key, value| value.dig(:inclusion, level, type) }
+
+      from_source(fields)
+    end
```

Comment:
Thanks for the review! I've made most of the changes.

So, part of this is me projecting my desire for it to look like ActiveRecord, but because they are keyword arguments, they have to be included, so really the signature is `find_by(attribute_name:)`.

However, I totally get your point, and I did wonder what would happen when somebody wanted to find by heading because `find_by(heading:)` wouldn't be allowed.

Reply:
Good point re: signature. I guess I typically try to make the method name descriptive enough in itself, but you're right that the keyword arguments are part of the signature. But as it stands, couldn't you switch the method names of `find_all_by` and `where`? That feels a little strange to me